actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -392,7 +392,7 @@ def route_source_location
             end
           else
             def route_source_location
-              if Mapper.route_source_locations
+              if Mapper.route_source_locations || ActionDispatch.verbose_redirect_logs
```

Comment:
If we are going to return route source locations, we might as well change 
how `Mapper.route_source_locations` is set and avoid Mapper having to know about `verbose_redirect_logs`.
We could set it here:
https://github.com/rails/rails/blob/d0f830033b2ea27bc1e66683ce5b16bd9ca55df1/actionpack/lib/action_dispatch/railtie.rb#L71

```ruby
      ActionDispatch::Routing::Mapper.route_source_locations = Rails.env.development? || ActionDispatch.verbose_redirect_logs
```

Or could we just leave it out?
Not sure what the result would be.